db/schema.rb

```diff
@@ -214,10 +217,13 @@
     t.integer "invitation_limit"
     t.integer "invited_by_id"
     t.string "invited_by_type"
+    t.datetime "deleted_at"
```

Comment:
It looks like you missed committing the migration that goes with this schema change.

Reply:
that schema change comes from a migration in the engine itself. the reason for that is the dependency that uses it is also part of the gem, not in Avalon itself. Does that make sense or would you prefer the migration be copied over?

Reply:
I've generally copied them over via the install:migration rake task so the db can always regenerate from the migrations.

Reply:
no problemo

Reply:
or maybe not - as long as the gem is installed these migrations will run.  if we copy them over rails gets upset because it then sees them twice.  I do think that not copying them is the better course here simply because if you have samvera-persona installed you get these migrations on any db:migrate. If you don't have samvera-persona installed (or comment it out) you never get these migrations. What do you think?

Reply:
I'll have to take a look at this tomorrow and give a quick ponder.

Reply:
I still think it would be cleaner and safer to follow the existing rails process of having the engine's migrations copied over via the install:migration rake task (https://edgeguides.rubyonrails.org/engines.html#engine-setup).  If we're concerned about the optional nature of this then I think It could be done as part of the persona install process.  I think devs would expect to see all of the migrations that led to the current schema in the db/migrations directory so keeping them in the gem is not transparent.  If persona is removed from the gemfile, then the migrations disappear from the migrations path but they still stay in the DB unless they were reversed.  We could chat about this synchronously if it would be helpful to resolve it.

Reply:
I can understand that.  I've basically been using this pattern - https://content.pivotal.io/blog/leave-your-migrations-in-your-rails-engines since 2011 or so, but I've expanded the code to handle both cases. That makes it nice for everyone as they can either include them or not as they see fit. I'll add the migrations back in on the Avalon side now and make sure it still runs.